src/unstract/sdk/adapters/url_validator.py

```
@@ -0,0 +1,172 @@
+            parsed = urlparse(url)
+
+            if not parsed.hostname:
+                return False, f"Invalid URL: No hostname found in '{url}'"
+
+            # Resolve hostname to IP address
+            try:
+                host_ip = socket.gethostbyname(parsed.hostname)
+            except socket.gaierror as e:
+                return (
+                    False,
+                    f"DNS resolution failed for '{parsed.hostname}': {str(e)}",
```

Ruff (0.12.2)

Use explicit conversion flag

Replace with conversion flag

(RUF010)